### ISSUE TITLE
Bug: Fatal Error in latest version of PHPUnit 8 (v8.5.27) on PHP 7.3.0

### ISSUE THREAD

**Issue by bshaffer** [sebastianbergmann/phpunit#5016]
| Q                   | A
| --------------------| ---------------
| PHPUnit version     | 8.5.27
| PHP version         | 7.3.0
| Installation Method | Composer

#### Summary

On PHP 7.3.0 / Ubuntu (note, this is not happening on 7.3.33 / MacOSX), I am getting the following fatal error when running phpunit. The tests are never executed. When I downgrade to `v8.5.26`, the tests run as expected.

```sh
$ vendor/bin/phpunit
PHPUnit 8.5.27 #StandWithUkraine

PHP Fatal error:  Uncaught TypeError: ReflectionClass::getMethods() expects parameter 1 to be int, null given in /workspace/vendor/phpunit/phpunit/src/Util/Reflection.php:45
Stack trace:
#0 /workspace/vendor/phpunit/phpunit/src/Util/Reflection.php(45): ReflectionClass->getMethods(NULL)
#1 /workspace/vendor/phpunit/phpunit/src/Util/Reflection.php(35): PHPUnit\Util\Reflection->filterMethods(Object(ReflectionClass), NULL)
#2 /workspace/vendor/phpunit/phpunit/src/Util/Test.php(513): PHPUnit\Util\Reflection->methodsInTestClass(Object(ReflectionClass))
#3 /workspace/vendor/phpunit/phpunit/src/Framework/TestSuite.php(551): PHPUnit\Util\Test::getHookMethods('Google\\Cloud\\Sa...')
#4 /workspace/vendor/phpunit/phpunit/src/Framework/TestSuite.php(619): PHPUnit\Framework\TestSuite->run(Object(PHPUnit\Framework\TestResult))
#5 /workspace/vendor/phpunit/phpunit/src/Te in /workspace/php-docs-samples/testing/vendor/phpunit/phpunit/src/Util/Reflection.php on line 45
```

**Comment by sebastianbergmann** [comment]
I am sorry but support for outdated versions such as 7.3.0 is not something I am able to provide.

**Comment by erdemkose** [comment]
I also have the same problem with PHP 7.2. And I think I have found the reason and in which PHP version it was fixed.

`ReflectionClass::getMethods()` allows `null` for the filter but until PHP 7.2.18 and 7.3.5, this is throwing a `TypeError` if you have `declare(strict_types=1);` which is the case for PHPUnit.

Nikita fixed it in PHP 7.2.18 and PHP 7.3.5 in a different context. Here you can find the related bug ticket: https://bugs.php.net/bug.php?id=77772.

PHP 7.2.17 - ReflectionClass::getMethods(null): https://3v4l.org/bCJPA#v7.2.17
```
Fatal error: Uncaught TypeError: ReflectionClass::getMethods() expects parameter 1 to be integer, null given in /in/bCJPA:11
Stack trace:
#0 /in/bCJPA(11): ReflectionClass->getMethods(NULL)
#1 {main}
  thrown in /in/bCJPA on line 11

Process exited with code 255.
```

PHP 7.2.18 - ReflectionClass::getMethods(null): https://3v4l.org/bCJPA#v7.2.18
```
Method [ <user> public method test ] {
  @@ /in/bCJPA 6 - 6
}
```
